subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -414,16 +351,50 @@ public void setGroovyClasspath(FileCollection groovyClasspath) {
      * @since 6.8
      */
     @Nested
-    @Optional
     public Property<JavaLauncher> getJavaLauncher() {
         return javaLauncher;
     }
 
+    /**
+     * Injects and returns an instance of {@link IncrementalCompilerFactory}.
+     *
+     * @since 5.6
+     */
+    @Inject
+    protected IncrementalCompilerFactory getIncrementalCompilerFactory() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
```suggestion
    protected abstract IncrementalCompilerFactory getIncrementalCompilerFactory();
```

💄 `GroovyCompile` is abstract now, so it makes sense to make all the injected stuff abstract as well.

Reply:
Good idea!